src/App.tsx

```diff
@@ -20,17 +25,114 @@ class App extends React.Component<{}, State> {
       './img/9.png',
       './img/10.png',
     ],
+    step: 3,
+    frameSize: 3,
+    itemWidth: 130,
+    animationDuration: 1000,
+    infinite: false,
   };
 
   render() {
-    const { images } = this.state;
+    const {
+      images, step, frameSize, itemWidth, animationDuration, infinite,
+    } = this.state;
+
+    const handleStep
+      = (event: React.ChangeEvent<HTMLInputElement>) => this.setState({
+        step: +event.target.value,
+      });
+    const handleFrameSize
+      = (event: React.ChangeEvent<HTMLInputElement>) => this.setState({
+        frameSize: +event.target.value,
+      });
+    const handleItemWidth
+      = (event: React.ChangeEvent<HTMLInputElement>) => this.setState({
+        itemWidth: +event.target.value,
+      });
+    const handleAnimationDuration
+      = (event: React.ChangeEvent<HTMLInputElement>) => this.setState({
+        animationDuration: +event.target.value,
+      });
+    const handleInfinite
+      = () => {
+        if (this.state.infinite === false) {
+          this.setState({
+            infinite: true,
+          });
+        } else {
+          this.setState({
+            infinite: false,
+          });
+        }
+      };
 
     return (
       <div className="App">
         {/* eslint-disable-next-line */}
-        <h1>Carousel with {images.length} images</h1>
+        <h1 data-cy="title">Carousel with {images.length} images</h1>
 
-        <Carousel />
+        <Carousel
+          images={images}
+          step={step}
+          frameSize={frameSize}
+          itemWidth={itemWidth}
+          animationDuration={animationDuration}
+          infinite={infinite}
+        />
+        <div className="input-container">
+          <label htmlFor="itemId">item - width </label>
+          <input
+            id="itemId"
+            defaultValue={itemWidth}
+            type="number"
+            min="100"
+            max="200"
+            onChange={handleItemWidth}
+            value={itemWidth}
+          />
+          <h1>
+            frame-size
+            <input
+              type="number"
+              min="1"
+              max="10"
+              onChange={handleFrameSize}
+              value={frameSize}
+            />
+          </h1>
+          <h1>
+            step
+            <input
+              type="number"
+              min="1"
+              max="5"
+              onChange={handleStep}
+              value={step}
+            />
+          </h1>
+          <h1>
+            animation-duration
+            <input
+              type="number"
+              min="100"
+              max="5000"
+              step="100"
+              onChange={handleAnimationDuration}
+              value={animationDuration}
+            />
+          </h1>
+          {this.state.infinite === false
```

Comment:
```suggestion
          {!this.state.infinite
```
always no need to compare boolean value with true/false because `===` operator return booleam too.
and here instead buttons use input with type `checkbox`